frontend/test/jest/tests/ProductListItem.test.tsx

```diff
@@ -0,0 +1,49 @@
+import { render, screen } from '@testing-library/react';
+import { ProductListItem } from '@components/product-list/sections/FilterableProductsSection/ProductList';
+import * as ProductSearch from '@components/product-list/sections/FilterableProductsSection/useProductSearchHitPricing';
+import { mockProduct } from 'test/jest/__mocks__/mockProduct';
+
+jest.mock('@ifixit/app');
+
+describe('ProductListItem', () => {
+   beforeEach(() => {
+      // @ts-ignore
+      ProductSearch.useProductSearchHitPricing = jest.fn().mockReturnValue({
+         price: 5,
+         compareAtPrice: 5,
+         isDiscounted: 5,
+         percentage: 5,
+      });
+   });
+
+   afterEach(() => {
+      jest.clearAllMocks();
+   });
+
+   it('renders and matches the snapshot', () => {
+      // @ts-ignore
+      const { asFragment } = render(<ProductListItem product={mockProduct} />);
+      const shortDescription = screen.getByText(
+         'Replace a dead or malfunctioning model EB-BG96aasd5ABE battery in a Samsung Galaxy S9 Plus smartphone.'
+      );
+
+      const reviewStars = screen.getByTestId('reviewStars');
+      (expect(reviewStars) as any).toBeInTheDocument();
+
+      (expect(shortDescription) as any).toBeInTheDocument();
+      (expect(asFragment()) as any).toMatchSnapshot();
+   });
+
+   it('renders without the review stars and matches the snapshot', () => {
```

Comment:
Do you think it's necessary to check all permutations?

```
rating < 4 && rating_count < 10 {
   shouldn't show stars
}
rating < 4 && rating_count > 10 {
   shouldn't show stars
}
raing > 4 && rating_count < 10 {
   shouldn't show starts
}
rating > 4 && rating_count > 10 {
   should show stars
}
```
Also, is it necessary to do a snapshot for this test?

Reply:
> Do you think it's necessary to check all permutations?

We definitely should! I think you caught a bug on master, [here](https://github.com/iFixit/react-commerce/blob/main/frontend/components/product-list/sections/FilterableProductsSection/ProductList.tsx#L145).
https://github.com/iFixit/react-commerce/blob/fb76f57c0e1e414db4cdf36da83e12879fd213c7/frontend/components/product-list/sections/FilterableProductsSection/ProductList.tsx#L145

We display the stars if the rating is greater than or equal to 4, **OR** when the rating count is greater than 10. 
This means we would end up displaying 2 stars if the review count is greater than 10. 

I'll clarify the expected behavior for displaying the stars and then make a follow-up pr to add all permutations. 

> Also, is it necessary to do a snapshot for this test? Meaning can we just check that the element is in the document instead of a whole snapshot test.

I don't think it's necessary to do a snapshot for this test. Removed the snapshot in https://github.com/iFixit/react-commerce/pull/438/commits/dc1403a39a423727539f9c5e798d5271210e6f06.

Reply:
>We display the stars if the rating is greater than or equal to 4, OR when the rating count is greater than 10.
This means we would end up displaying 2 stars if the review count is greater than 10.

Oh, nevermind. I think it is the expected behavior. We will show any review stars if the review count is greater than 10.
Found it from here https://github.com/iFixit/react-commerce/issues/215:
 
> Don't show the reviews in the product container if the average rating is < 4 stars unless the review count is > 10.

Reply:
Addressed in https://github.com/iFixit/react-commerce/pull/438/commits/63d6578415daf3f5e739a5a434d308aa1b41d8a1.